hathor/manager.py

```diff
@@ -506,19 +516,33 @@ def generate_parent_txs(self, timestamp: Optional[float]) -> 'ParentTxs':
         This method tries to return a stable result, such that for a given timestamp and storage state it will always
         return the same.
         """
-        if timestamp is None:
-            timestamp = self.reactor.seconds()
-        can_include_intervals = sorted(self.tx_storage.get_tx_tips(timestamp - 1))
-        assert can_include_intervals, 'tips cannot be empty'
-        max_timestamp = max(int(i.begin) for i in can_include_intervals)
-        must_include: List[bytes] = []
-        assert len(can_include_intervals) > 0, f'invalid timestamp "{timestamp}", no tips found"'
-        if len(can_include_intervals) < 2:
-            # If there is only one tip, let's randomly choose one of its parents.
-            must_include_interval = can_include_intervals[0]
-            must_include = [must_include_interval.data]
-            can_include_intervals = sorted(self.tx_storage.get_tx_tips(must_include_interval.begin - 1))
-        can_include = [i.data for i in can_include_intervals]
+        # get all that are before "timestamp"
+        # XXX: maybe add some tolerance?
+        tx_tips = list(self.tx_storage.iter_tx_tips(timestamp))
```

Comment:
This algorithm seems good but I feel we haven't tested this new index enough. Maybe we should do this change in the next PR so we can test this index more before using it in such critical method.

Reply:
I'll move it to the next PR.

Reply:
Moved to the next PR. Thanks!